plugins/processors/aws/ec2/ec2.go

```diff
@@ -71,6 +87,21 @@ func (r *AwsEc2Processor) Add(metric telegraf.Metric, _ telegraf.Accumulator) er
 	return nil
 }
 
+func (r *AwsEc2Processor) logCacheStatistics() {
+	if r.tagCache != nil {
+		for {
+			r.Log.Debugf("cache: size=%d hit=%d miss=%d full=%d\n",
+				r.tagCache.EntryCount(),
+				r.tagCache.HitCount(),
+				r.tagCache.MissCount(),
+				r.tagCache.EvacuateCount(),
+			)
+			r.tagCache.ResetStatistics()
+			time.Sleep(30 * time.Second)
+		}
+	}
```

Comment:
How about return early on
```golang
	if r.tagCache == nil {
		return
	}
```

Furthermore, I would really love to see a `time.Ticker` here instead of sleep... And, last but not least, I don't see how you are able to leave the for loop when Telegraf is stopped. You could do this via a context or similar.

Reply:
@srebhan this might be my general lack of Go knowledge, but I assume you would leave it when you stop telegraf. as in, telegraf is no longer running, so that loop should just no longer be running. these plugins don't run as subprocesses of the telegraf process, do they?

and the processors are "always" running so long as there are metrics being received, right?

would appreciate any further guidance you can provide

Reply:
i had some assistance from a peer, lemme send an updated PR. and you can give feedback

Reply:
@caphrim007 both fine with me. If you use a ticker you could do something along the lines of

```golang
for {
    select {
        case  <-ticker.C:
            ....
        case <-ctx.Done():
            return
    }
}
```

Usually a context is much easier to use for such cases...